### ISSUE TITLE
[BUG] Description

### ISSUE THREAD

**Issue by damzog** [OpenAPITools/openapi-generator#18459]
##### Description

Generated Python code fails to deserialize dictionary objects

##### openapi-generator version

7.5.0

##### OpenAPI declaration file content or url

see [this repo](https://github.com/damzog/openapi-generater-issue/tree/master/openapi_client) for complete example and explanation.

##### Generation Details

Problem is probably caused by the code to deserialize a dict:

```python
    @classmethod
    def from_dict(cls, obj: Dict) -> Self:
        """Create an instance of User from a dict"""
        if obj is None:
            return None
        # Should probably be 
        # if isinstance(obj, dict)
        # but is
        if not isinstance(obj, dict):
            return cls.model_validate(obj)

        _obj = cls.model_validate({
            "name": obj.get("name"),
            "address": Address.from_dict(obj.get("address")) if obj.get("address") is not None else None,
        })
        return _obj
```
##### Steps to reproduce

Checkout [this repo](https://github.com/damzog/openapi-generater-issue/tree/master/openapi_client)
pip install -r requirements.txt
uvicorn main:app --reload
python client.py

**Comment by wing328** [comment]
so if you remove/comment out the following:

```
        if not isinstance(obj, dict):
            return cls.model_validate(obj)
```

the code works as expected?

**Comment by damzog** [comment]
@wing328 no. In my case all obj passed to this class are instance of dict so this code is anyway noop. It does work if you omit the "not" meaning if you simply return  return cls.model_validate(obj) . In that case pydantic will deserialize (one might to also need to assign the result to _obj)

**Comment by damzog** [comment]
And as you can see
'''
  _obj = cls.model_validate({
            "name": obj.get("name"),
            "address": Address.from_dict(obj.get("address")) if obj.get("address") is not None else None,
        })
'''
filters out all entries from a dict that are not name, address which I do not understand why (sorry code snippet is not completely up to date. I removed address and added age in the example repo but it does not make a difference.)

**Comment by AlexGuzmanAtAbsa** [comment]
I came across the same issue while using the python generator on the latest version. In short, the crux of the problem is the fact the generated validation code omits to generate validation code for map properties. in other words:
```
_obj = cls.model_validate({
...
})
```
 
Would omit any entries for mandatory properties what were originally defined as a map on the server side.

Happy to provide more details if necessary but clearly the problem is to do with handling of mapping types.

**Comment by AlexGuzmanAtAbsa** [comment]
I wanted to add a folllow-up note.After addressing some OpenAPI JSON generation issues on a Scala server  solved the issue.(Specifically,  lack of type attribute in the generated spec)

The actual issue had to do with Swagger rendering incorerct JSON out the Components object tree (it removes the type attribute from the docs enpoint even though the underlying Components object is correct).Serializing and deserializng the Componnets object and then passing this again to the OpenApi object solved the issue. After this the client generator generated the correct validation for Map types.
